Add support for thousands separators in [bigint] casting

h4vox opened the pull request:
# PR Summary

This PR resolves a parsing issue in PowerShell where numeric strings containing thousands separators (e.g., `1,000`) fail to convert into `[bigint]`, resulting in an `InvalidArgument` exception. The root cause is the absence of `NumberStyles.AllowThousands` and lack of formatting cleanup before conversion. This update ensures consistent and locale-aware conversion behavior for large numeric strings.

## PR Context

The issue occurs when users attempt to cast strings like `'1,000'`  to `[bigint]`, and PowerShell does not account for grouping symbols defined by culture. This leads to type conversion errors in scripts or user input relying on formatted numbers.

## Solution
Updates made to the `ConvertFrom()` method in `LanguagePrimitives.cs` include:

- Casting `sourceValue` using `(string)` instead of `as string` for type safety.
- Dynamically handling culture-specific formatting to ensure group separators are correctly removed.
- Including `NumberStyles.AllowThousands` for accurate parsing of `BigInteger`.

### Error Message Before Fix

```Powershell

InvalidArgument: Cannot convert value "1,000" to type "System.Numerics.BigInteger". Error: "The value could not be parsed."

```
### After Fix

```Powershell

PS> [bigint] '1,000'
'1000'

```
### Why This Fix?

- Prevents casting errors when handling numeric strings with thousands separators.
- Improves PowerShell’s type conversion reliability.
- Ensures compatibility with standard numeric formatting.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge. If this PR is a work in progress, please open this as a [Draft Pull Request and mark it as Ready to Review when it is ready to merge](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [x] None

- **User-facing changes**
  - [x] Not Applicable

- **Testing - New and feature**
  - [x] added a new test (https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - Added unit tests to verify comma-separated numeric string conversion.
    
## Related Issues 

## FIXED ---- 
- Fixes #25368

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors.

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -199,8 +199,14 @@ public override bool CanConvertFrom(object sourceValue, Type destinationType)
+            string sourceAsString = LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider) as string;
+            
+                if (sourceAsString != null){
+                    sourceAsString = sourceAsString.Replace(",", ""); // Remove commas before conversion
```
Digital group symbol, or `NumberGroupSeparator`, depends on current culture:
```powershell
(Get-Culture).NumberFormat | fl NumberGroupSeparator
```

AbishekPonmudi replied:
Hi @iSazonov,

Thanks for highlighting the culture-specific nature of `NumberGroupSeparator`. I agree that hardcoding the comma (`,`) limits the fix to cultures like en-US and won’t handle other separators (e.g., `.` in de-DE or spaces in some locales).

To address this, I suggest modifying the code to use `NumberGroupSeparator` from the `formatProvider` or the current culture. Here’s the proposed update:

```
string sourceAsString = LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider) as string;
if (sourceAsString != null)
{
    string groupSeparator = formatProvider?.GetFormat(typeof(NumberFormatInfo)) is NumberFormatInfo nfi 
        ? nfi.NumberGroupSeparator 
        : CultureInfo.CurrentCulture.NumberFormat.NumberGroupSeparator;
    sourceAsString = sourceAsString.Replace(groupSeparator, string.Empty); // Remove culture-specific group separator
}
return LanguagePrimitives.ConvertTo(sourceAsString, destinationType, formatProvider);
```
This change:

Retrieves the NumberGroupSeparator from the formatProvider if available, or falls back to CultureInfo.CurrentCulture.

Removes the culture-specific group separator before conversion, ensuring compatibility across locales.

I will also add unit tests to verify the fix for different cultures (e.g., en-US, de-DE). Does this approach look good, or do you have additional suggestions? Let me know, and I can update the PR accordingly!

AbishekPonmudi replied:
Hi @iSazonov, thanks for the feedback! I’ve updated `LanguagePrimitives.cs` to use `NumberGroupSeparator` for culture-specific handling and pushed the changes. Please review and let me know if it’s good!

h4vox committed:
Merge branch 'master' into patch-3

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (#25396)

# Issue Summary
PowerShell fails to convert numeric strings with thousands separators (e.g., commas) to `[bigint]`, causing casting errors. This occurs because the `ConvertFrom()` method in `LanguagePrimitives.cs` doesn’t handle formatted numbers.

## Cause
- The `ConvertTo()` method doesn’t remove thousands separators (e.g., `,` in en-US), leading to `BigInteger.Parse()` failures.
- This affects scripts handling large numbers with separators.

## Solution
Modify `ConvertFrom()` to:
- Detect the culture-specific thousands separator.
- Remove it before conversion.

iSazonov commented:
In related issue there is a reference to code where exception is raised https://github.com/PowerShell/PowerShell/blob/dd1fb9167f6388ba4911478940ce8615dd61f5ef/src/System.Management.Automation/engine/LanguagePrimitives.cs#L2946-L2955
I guess it is right place for a fix. 
Also I don't think we should do any normalizations since BigInteger.(Try)Parser() can do all work.

AbishekPonmudi commented:
Hey @iSazonov,

Thanks for the heads-up! I went ahead and tweaked `LanguagePrimitives.cs` (lines 2957–2966) to fix that `BigInteger.Parse()` issue with thousands separators. I ditched `NumberStyles.AllowHexSpecifier`, swapped `BigInteger.Parse()` for `BigInteger.TryParse()` to keep things smooth, and made sure it handles stuff like "1,000" → 1000 and "5,432,100" → 5432100 correctly.

```csharp
if (resultType == typeof(BigInteger)) 
{ 
    // Adjust NumberStyles to remove AllowHexSpecifier for standard numeric parsing
    NumberStyles style = NumberStyles.AllowLeadingSign 
        | NumberStyles.AllowDecimalPoint 
        | NumberStyles.AllowExponent; // Removed AllowHexSpecifier  

    BigInteger parsedValue;
    if (BigInteger.TryParse(strToConvert, style, NumberFormatInfo.InvariantInfo, out parsedValue))
    {
        return parsedValue;
    }
    throw new PSInvalidCastException("Failed to convert string to BigInteger.");
}

h4vox committed:
Fix: Remove AllowHexSpecifier in BigInteger parsing to support formatted numeric strings (#25396)

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -2954,17 +2954,21 @@ private static object ConvertStringToInteger(
-                    return BigInteger.Parse(strToConvert, style, NumberFormatInfo.InvariantInfo);
+               if (resultType == typeof(BigInteger))
+               { 
+                    // Adjust NumberStyles to remove AllowHexSpecifier for standard numeric parsing
```
Please remove obvious comment.

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -2954,17 +2954,21 @@ private static object ConvertStringToInteger(
+                        return parsedValue;
+                    }
+                    
+                    throw new PSInvalidCastException("Failed to convert string to BigInteger.");
```
The code is in try-catch block where we create generic error message. So we have no need re-throw and can use BigInteger.Parse() instead of BigInteger.TryParse().

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -2954,17 +2954,21 @@ private static object ConvertStringToInteger(
+               if (resultType == typeof(BigInteger))
+               { 
+                    // Adjust NumberStyles to remove AllowHexSpecifier for standard numeric parsing
+                    NumberStyles style = NumberStyles.AllowLeadingSign 
+                    | NumberStyles.AllowDecimalPoint 
+                    | NumberStyles.AllowExponent; 
```
Since BigInteger is an [integer](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Globalization/NumberStyles.cs,63) and we want to allow thousands separator I'd consider:
```c#
NumberStyles style = NumberStyles.Integer+NumberStyles.AllowThousands;
```

AbishekPonmudi replied:
Thanks for the review, @iSazonov! I’ve updated LanguagePrimitives.cs to remove the unnecessary comment, replaced BigInteger.TryParse() with BigInteger.Parse(), and adjusted NumberStyles to allow thousands separators. Let me know if any further refinements are needed!

h4vox committed:
Fix: Update BigInteger parsing to use Parse() and support thousands separators (#25396)

h4vox committed:
Fix: Update BigInteger parsing to use Parse() and support thousands separators (#25396)

h4vox reviewed (commented):
Thanks for the review, @iSazonov! I’ve updated LanguagePrimitives.cs to remove the unnecessary comment, replaced BigInteger.TryParse() with BigInteger.Parse(), and adjusted NumberStyles to allow thousands separators. Let me know if any further refinements are needed!

iSazonov commented:
@AbishekPonmudi Please add new tests to cover the code. (The tests should use current culture NumberGroupSeparator .)

h4vox committed:
Merge branch 'master' into patch-3

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -199,8 +199,19 @@ public override bool CanConvertFrom(object sourceValue, Type destinationType)
         public override object ConvertFrom(object sourceValue, Type destinationType, IFormatProvider formatProvider, bool ignoreCase)
         {
-            string sourceAsString = (string)LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider);
+            // Fix: Remove commas from numeric strings before conversion to prevent casting errors.
+            string sourceAsString = LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider) as string;
+            
+            if (sourceAsString != null){
+            
+                string groupSeparator = formatProvider?.GetFormat(typeof(NumberFormatInfo)) is NumberFormatInfo nfi 
+                    ? nfi.NumberGroupSeparator 
+                    : CultureInfo.CurrentCulture.NumberFormat.NumberGroupSeparator;
+                    
+                sourceAsString = sourceAsString.Replace(groupSeparator, string.Empty); // Remove culture-specific group separator
+            }
             return LanguagePrimitives.ConvertTo(sourceAsString, destinationType, formatProvider);
+
```
I wonder why you have added the code again.

AbishekPonmudi replied:
"Thanks for your review, @iSazonov! I implemented the NumberGroupSeparator logic to ensure numeric conversion works correctly across different locales. If this logic was already handled elsewhere and is redundant, let me know, and I can adjust accordingly!"

iSazonov replied:
There is BigInteger.Parse() below where you corrected style and which does all work we need.

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (PowerShell#25396)

h4vox committed:
Remove commas from numeric strings before conversion to prevent casting error (Powershell #25396)

h4vox reviewed (commented):
Thanks for the review, @iSazonov! I’ve updated the code to remove manual separator normalization and now rely on BigInteger.Parse() with NumberStyles.AllowThousands to handle formatted numeric strings directly. Let me know if any further refinements are needed!

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (#25396)

iSazonov commented:
@AbishekPonmudi Please add tests to cover all scenarios you are trying to fix.

AbishekPonmudi commented:
> @AbishekPonmudi Please add tests to cover all scenarios you are trying to fix.

![TEST COMPLETED](https://github.com/user-attachments/assets/66413e4b-9d54-4a3e-8d98-b0d98a07f227)

Hey @iSazonov,

I ran manual tests for different locales using the file `BigIntegerCultureHandling.ps1` as mentioned in the visual reference. I verified that PowerShell correctly handles thousands of separators, everything seems to be working fine based on these tests.  
Please let me know if there are any issues or if any further refinements are needed :)

iSazonov commented:
@AbishekPonmudi You should add tests to the PR.

iSazonov commented:
@AbishekPonmudi If the PR fix https://github.com/PowerShell/PowerShell/issues/25368 we need tests in the PR which cover scenarios from that issue.

h4vox committed:
Merge branch 'PowerShell:master' into patch-3

h4vox committed:
Tests for BigInteger Parsing Across Cultures (#25396)

h4vox committed:
Tests for BigInteger Parsing Across Cultures (#25396)

AbishekPonmudi commented:
Hi @iSazonov,  

I have updated the test cases in **PR #25396** to fully cover the scenarios described in **Issue #25368**. These tests validate PowerShell’s type conversion via `LanguagePrimitives.ConvertTo()`, ensuring correct handling of thousands separators across cultures.  

The test case is included here: [Test Case in PR #25396](https://github.com/PowerShell/PowerShell/pull/25396/commits/1a4cc3f295b3a9cb858b43c6474f97dae91db958).  

i also attached the testcase that fix the exact problem on **Issue #25368**.
![final- tescase issue fixed](https://github.com/user-attachments/assets/4cb0572c-373c-4576-878d-d11fc0811f90)

![image](https://github.com/user-attachments/assets/623bee07-2e96-4154-bea0-1944c0c1edfa)

Please review and let me know if any further refinements are needed :)

h4vox committed:
Tests for BigInteger Parsing Across Cultures (#25396)

h4vox reviewed (commented):
Hi @iSazonov,  

I have updated **PR #25396** based on the feedback and ensured that it includes tests covering **Issue #25368** scenarios.  
Please take a look and let me know if any further refinements are needed.

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -199,10 +199,14 @@ public override bool CanConvertFrom(object sourceValue, Type destinationType)
-
+            string sourceAsString = LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider) as string;
+                
+                if (!string.IsNullOrEmpty(sourceAsString) && destinationType == typeof(BigInteger)) 
+                {
+                    return BigInteger.Parse(sourceAsString, NumberStyles.Integer | NumberStyles.AllowThousands, NumberFormatInfo.InvariantInfo);
+                }
```
@AbishekPonmudi Could you please explain why do need the new code? What test cover this?

AbishekPonmudi replied:
Hi @iSazonov, This update improves `[bigint]` conversion by allowing PowerShell to handle thousands separators (`1,000`, `1,00,000`) directly using `NumberStyles.AllowThousands`, removing manual preprocessing.  

Previously, PowerShell required **manual separator removal**, which was **culture-dependent** and added extra processing. The updated approach simplifies this by leveraging `BigInteger.Parse()` with `NumberStyles.AllowThousands`, ensuring formatted numbers—including **large values like `9223372036854775,807`** are handled **automatically**.  

Let me know if any further refinements are needed!

h4vox committed:
Merge branch 'master' into patch-3

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -2943,17 +2947,12 @@ private static object ConvertStringToInteger(
+               if (resultType == typeof(BigInteger))
+               { 
+                    NumberStyles style = NumberStyles.Integer | NumberStyles.AllowThousands;
+                    
```
All test works with only change in line 2952. So we can remove all you added in lines 204-207.

AbishekPonmudi replied:
I agree with your point, are you suggesting that I should **remove lines 204-207 entirely**, or would it make more sense to **adjust them** instead? Just want to confirm before making the changes.

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (PowerShell#25396)

h4vox reviewed (commented):
I agree with your point, I've removed **lines 204-207** as suggested and reverted to the previous implementation. Everything should now align with the existing logic in **line 2952**, ensuring proper handling of thousands separators.  

Could you clarify if any issues remain with this change, or if further adjustments are needed?

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -199,10 +199,9 @@ public override bool CanConvertFrom(object sourceValue, Type destinationType)
-            string sourceAsString = (string)LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider);
-            return LanguagePrimitives.ConvertTo(sourceAsString, destinationType, formatProvider);
+           string sourceAsString = (string)LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider);
+           return LanguagePrimitives.ConvertTo(sourceAsString, destinationType, formatProvider);
```
Unrelated change.

AbishekPonmudi replied:
Could you clarify which part you’re referring to as an unrelated change? I removed **lines 204-207** and reverted to the previous implementation, aligning everything with **line 2952** for proper handling of thousands separators.  

I just want to ensure everything is correct. Let me know, and I’ll adjust if needed

iSazonov replied:
Indentation. You removed space in the lines.

AbishekPonmudi replied:
I fixed the indentation issue and removed **lines 204-207** as suggested. I also verified the test cases to ensure proper handling.

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (PowerShell#25396)

iSazonov commented on src/System.Management.Automation/engine/LanguagePrimitives.cs:
```diff
@@ -199,10 +199,9 @@ public override bool CanConvertFrom(object sourceValue, Type destinationType)
         public override object ConvertFrom(object sourceValue, Type destinationType, IFormatProvider formatProvider, bool ignoreCase)
         {
-            string sourceAsString = (string)LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider);
+            string sourceAsString = LanguagePrimitives.ConvertTo(sourceValue, typeof(string), formatProvider) as string;
```
Please revert to original code.

h4vox committed:
Fix: Remove commas from numeric strings before conversion to prevent casting errors (PowerShell#25396)

iSazonov commented:
@mklement0 Could you please review? Does this fix your issue? Should we add more tests?

AbishekPonmudi commented:
@iSazonov, are there any remaining issues preventing this PR from being merged? Let me know if anything else needs to be addressed!

mklement0 commented:
Thanks for tackling this, @AbishekPonmudi.

* The code LGTM.

* As for the tests:
  * I think that one culture-specific test should be enough, using a culture where `,` is the decimal mark and `.` is the thousands separator (i.e. the opposite of the invariant / en-US culture), e.g. `de-DE`. Due to the intentional use of the invariant culture, strings such as `'1,000'` and `'1.200'` should still parse as `1000` and `1` when using a `[bigint]` cast.
  Incidentally, do we have such tests for other types?

  * I would add a test for casting from a floating-point number strings, to verify that the fallback via `[double]` still works, e.g. `[bigint] '1.2'` should yield `1`

Incidentally - and this behavior predates the PR and may warrant a new bug report - why is `[bigint] '1.9'` currently `1`, not `2`, as it is with all the primitive integer types (e.g. `[int] '1.9'`)?

AbishekPonmudi commented:
Thanks for the suggestion! I’ve reviewed the points, and here’s my take:

I agree with your suggestion to add a culture-specific test. Specifically, for a culture where , is the decimal mark and . is the thousands separator (e.g., de-DE). This will help ensure we handle culture-specific parsing correctly.
```
It 'Correctly parses using de-DE culture with thousands separator as dot' {
    $result = [System.Numerics.BigInteger]::Parse("1.000", [System.Globalization.NumberStyles]::Number, [System.Globalization.CultureInfo]::GetCultureInfo("de-DE"))
    $result | Should -Be 1000
}
```
In this test, "1.000" in de-DE parses as 1000, because in this culture, . is the thousands separator, unlike the invariant culture (en-US), where . is interpreted as a decimal point.

##Important Note:## The test case here uses **BigInteger.Parse** directly to show culture-specific parsing. However, it’s important to note that when casting with `[bigint]`, PowerShell uses the **InvariantCulture**. So in normal `[bigint]` conversion, `"1.000"` would still `return 1` as the invariant culture treats the . as a decimal separator. Therefore, we may need to implement a culture-aware conversion if we want this behavior for `[bigint]`.

##Next## 

I’ve also added a test to verify the fallback behavior when casting floating-point number strings to a `bigint`. This ensures that PowerShell properly casts `"1.2"` to `1`, handling the fractional part by truncation.

```` It 'Casts floating-point string to bigint and truncates the decimal' {
    $result = [bigint] "1.2"
    $result | Should -Be 1
} ```

h4vox committed:
Tests for BigInteger Parsing Across Cultures (PowerShell#25396)

h4vox committed:
Merge branch 'master' into patch-3

mklement0 commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,51 @@
+
+Describe 'PowerShell Type Conversion - BigInteger Parsing' -Tag 'CI' {
+
+    It 'Can convert formatted numbers using PowerShell type system' {
```
Thanks for updating. I'm thinking we don't need this and the "hi-IN" test: the new "de-DE" test is sufficient to demonstrate that the behavior is culture-invariant, as desired.

mklement0 commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    It 'Parses number using de-DE culture but falls back to invariant behavior' {
+        [System.Globalization.CultureInfo]::CurrentCulture = [System.Globalization.CultureInfo]::GetCultureInfo("de-DE")
```
To be save, I suggest saving and restoring the original `[System.Globalization.CultureInfo]::CurrentCulture` value. Also note that type accelerator `[cultureinfo]` is available.

Other than that: LGTM overall now.

h4vox committed:
Tests for BigInteger Parsing Across Cultures (PowerShell#25396)

h4vox committed:
Merge branch 'master' into patch-3

mklement0 commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,50 @@
+Describe 'PowerShell Type Conversion - BigInteger Parsing' -Tag 'CI' {
+
+    It 'Can convert formatted numbers using PowerShell type system' {
+        [System.Globalization.CultureInfo]::CurrentCulture = [System.Globalization.CultureInfo]::GetCultureInfo("en-US")
```
Since the culture angle is now covered by the `de-DE` test, I suggest removing this.

mklement0 commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,50 @@
+        $convertedValue | Should -Be 1000
+    }
+
+    It 'Handles large comma-separated numbers that previously failed' {
```
Quibble: perhaps better: `Handles large numbers with thousands separators that previously failed'

mklement0 commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,50 @@
+        $convertedValue | Should -Be 1000000
+    }
+
+    It 'Parses number using de-DE culture but falls back to invariant behavior' {
```
Quibble: Since the current culture never comes into play, I wouldn't say "Parses number using de-DE". Perhaps better: `Parses a number string using the invariant culture, irrespective of the current culture`

h4vox committed:
Tests for BigInteger Parsing Across Cultures (PowerShell#25396)

h4vox committed:
Tests for BigInteger Parsing Across Cultures (PowerShell#25396)

mklement0 reviewed (approved):
@iSazonov, everything LGTM now.

iSazonov commented on test/powershell/engine/BigIntegerCultureHandling.test.ps1:
```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'PowerShell Type Conversion - BigInteger Parsing' -Tag 'CI' {
```
Please all tests in LanguagePrimitive.Tests.ps1

AbishekPonmudi replied:
@iSazonov Could you clarify exactly what needs to be done with the test cases?  
Should I move all existing tests from `BigIntegerCultureHandling.test.ps1` into `LanguagePrimitive.Tests.ps1`, or just specific ones?

h4vox committed:
Add Tests for BigInteger Parsing Across Cultures (#25396)

h4vox committed:
Delete test/powershell/engine/BigIntegerCultureHandling.test.ps1

h4vox committed:
Add Tests for BigInteger Parsing Across Cultures (#25396)

iSazonov commented on test/powershell/engine/Api/LanguagePrimitive.Tests.ps1:
```diff
@@ -185,4 +185,49 @@ Describe "Language Primitive Tests" -Tags "CI" {
         $test.TestHandlerReturnObject() | Should -BeTrue
     }
+
+    It 'Can convert formatted numbers using PowerShell type system' {
+        [System.Globalization.CultureInfo]::CurrentCulture = [System.Globalization.CultureInfo]::GetCultureInfo("en-US")
+        $formattedNumber = "1,000"
+        $convertedValue = [System.Management.Automation.LanguagePrimitives]::ConvertTo($formattedNumber, [bigint])
+        $convertedValue | Should -Be 1000
+    }
```
The test was removed before.

h4vox committed:
Add Tests for BigInteger Parsing Across Cultures (#25396)

AbishekPonmudi commented:
@iSazonov Is everything is resolved.

iSazonov commented:
/azp PowerShell-CI-linux-packaging,PowerShell-Windows-Packaging-CI

iSazonov commented:
/azp run PowerShell-CI-linux-packaging,PowerShell-Windows-Packaging-CI

iSazonov commented:
@mklement0 Thanks for review!

AbishekPonmudi commented:
> @AbishekPonmudi Thanks for your contribution! Please update the PR description to exactly reflect the change.

@iSazonov Thank you for reviewing my contribution! I appreciate your feedback. I have updated the PR description to reflect the implemented changes.
